### PR TITLE
Resolve v_prediction issue for min-SNR gamma weighted loss function

### PR DESCRIPTION
**Pull request by bghira**
# What does this PR do?

Flooring the SNR weights to 1 results in a solution for the Inf position that arises with a zero-terminal SNR model.

As such, we can safely remove this check and the parameter can be even used by default with the value of 5.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by bghira** [comment]
the first curve is a normal loss calc. and then, followed by using snr gamma value of 5 while training v_prediction and zero-terminal SNR SDXL:

image

i've now trained with this for about 6,000 steps and it's great. works very well.

File: examples/text_to_image/train_text_to_image_sdxl.py [hunk 1]
```diff
@@ -1008,9 +987,27 @@ def compute_time_ids(original_size, crops_coords_top_left):
...
-                    )
+                    if noise_scheduler.config.prediction_type == "epsilon":
+                        mse_loss_weights = (
+                            torch.stack(
+                                [snr, args.snr_gamma * torch.ones_like(timesteps)],
+                                dim=1,
+                            ).min(
+                                dim=1
+                            )[0]
+                            / snr
```
Comment (sayakpaul) [inline examples/text_to_image/train_text_to_image_sdxl.py:998]: Could we please break it down in variables?

File: examples/text_to_image/train_text_to_image_sdxl.py [hunk 2]
```diff
@@ -1008,9 +987,27 @@ def compute_time_ids(original_size, crops_coords_top_left):
...
+                            + 1
+                        )
```
Comment (sayakpaul) [inline examples/text_to_image/train_text_to_image_sdxl.py:1010]: Same.

**Review by sayakpaul — APPROVED** [review]
Good to merge when the comments are addressed. 

Thanks for spotting and fixing.

**Review by sayakpaul — APPROVED** [review]
Much better readability now. Thanks much!

**Comment by bghira** [comment]
@sayakpaul ah, i'm used to pre-commit hooks doing this for me :D done

**Comment by sayakpaul** [comment]
Failing tests are flaky!

File: examples/text_to_image/train_text_to_image_sdxl.py [hunk 3]
```diff
@@ -1013,9 +992,17 @@ def compute_time_ids(original_size, crops_coords_top_left):
...
                     # This is discussed in Section 4.2 of the same paper.
                     snr = compute_snr(timesteps)
-                    mse_loss_weights = (
+                    base_weight = (
                         torch.stack([snr, args.snr_gamma * torch.ones_like(timesteps)], dim=1).min(dim=1)[0] / snr
                     )
+
+                    if noise_scheduler.config.prediction_type == "v_prediction":
+                        # Velocity objective needs to be floored to an SNR weight of one.
+                        mse_loss_weights = base_weight + 1
```
Comment (patrickvonplaten) [inline examples/text_to_image/train_text_to_image_sdxl.py:1001]: Why don't we do:
```py
mse_loss_weights = max(base_weight, 1) 
```

if we want to "floor" the SNR weight?

Comment (bghira) [reply examples/text_to_image/train_text_to_image_sdxl.py:1001]: `max` would be `ceil`, here we're just adding one to ensure a value of zero is a minimum of 1. we don't want to blindly overwrite minimal values with 1's, we want to scale all of them.

Comment (patrickvonplaten) [reply examples/text_to_image/train_text_to_image_sdxl.py:1001]: `mse_loss_weights = max(base_weight, 1)` in that case works more like floor - e.g. `mse_loos_weights` is equal to at least 1 or higher. 

Can you explain in a bit more detail why we want to add 1 here all the time?

Comment (bghira) [reply examples/text_to_image/train_text_to_image_sdxl.py:1001]: certainly. when using v-prediction, there is an infinite SNR at the final timestep. if you do not floor the weight of this to 1, it will lead to a NaN loss value and the model fails to produce useable predictions.

this was observed in the original min-SNR paper, and has since been implemented in the reference code upstream.
